components/permissions/brave_permission_manager.cc

```diff
@@ -0,0 +1,31 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/permissions/brave_permission_manager.h"
+
+#include <utility>
+
+#include "components/permissions/permission_context_base.h"
+
+namespace permissions {
+
+BravePermissionManager::BravePermissionManager(
+    content::BrowserContext* browser_context,
+    PermissionContextMap permission_contexts)
+    : PermissionManager(browser_context, std::move(permission_contexts)) {}
+
+GURL BravePermissionManager::GetCanonicalOrigin(
+    ContentSettingsType permission,
+    const GURL& requesting_origin,
+    const GURL& embedding_origin) const {
+  // Use requesting_origin which will have ethereum address info.
+  if (permission == ContentSettingsType::BRAVE_ETHEREUM)
+    return requesting_origin;
+
+  return PermissionManager::GetCanonicalOrigin(permission, requesting_origin,
```

Comment:
use PermissionsClient::OverrideCanonicalOrigin instead

Reply:
`GetCanonicalOrigin` is called in https://source.chromium.org/chromium/chromium/src/+/main:components/permissions/permission_manager.cc;l=373;drc=4d354dd4d0fa1db2f924262a4abe67e2fda9d7b4;bpv=1;bpt=1 where each of my sub-requests's requesting_origin (for each address) will be changed to embedding_origin if we don't override this.